[GStreamer][VideoCapture] Remove video/x-raw mime-type caps filters

jameshilliard opened the pull request:
#### 696df7229c41ce048133698debfb9faea81a4d08
<pre>
[GStreamer][VideoCapture] Remove video/x-raw mime-type caps filters
<a href="https://bugs.webkit.org/show_bug.cgi?id=242256">https://bugs.webkit.org/show_bug.cgi?id=242256</a>

Reviewed by Philippe Normand.

We support image/jpeg mime-type sources so having a video/x-raw
mime-type caps filter doesn&apos;t seem correct.

* Source/WebCore/platform/mediastream/gstreamer/GStreamerCaptureDeviceManager.cpp:
(WebCore::GStreamerCaptureDeviceManager::refreshCaptureDevices):

Canonical link: <a href="https://commits.webkit.org/252103@main">https://commits.webkit.org/252103@main</a>
</pre>

philn commented on Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoCapturer.cpp:
```diff
@@ -82,9 +82,6 @@ GstElement* GStreamerVideoCapturer::createConverter()
         head = m_videoSrcMIMETypeFilter.get();
 
-        auto caps = adoptGRef(gst_caps_new_empty_simple("video/x-raw"));
-        g_object_set(m_videoSrcMIMETypeFilter.get(), "caps", caps.get(), nullptr);
```
If we never set caps on this the whole element is kinda useless.

jameshilliard replied:
Isn't it getting set [here](https://github.com/WebKit/WebKit/blob/92cc4d5d99140da8ed1154a4eb309eba647841d9/Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoCapturer.cpp#L359) as well?

jameshilliard replied:
Best I can tell setting `video/x-raw` here doesn't make much sense as a plain `video/x-raw` mime-type isn't even valid without a format specifier from my understanding.

jameshilliard replied:
Removed this part for now since it's not really clear what the correct fix is.

philn commented on Source/WebCore/platform/mediastream/gstreamer/GStreamerCaptureDeviceManager.cpp:
```diff
@@ -135,8 +135,7 @@ void GStreamerCaptureDeviceManager::refreshCaptureDevices()
         case CaptureDevice::DeviceType::Camera: {
-            auto caps = adoptGRef(gst_caps_new_empty_simple("video/x-raw"));
-            gst_device_monitor_add_filter(m_deviceMonitor.get(), "Video/Source", caps.get());
+            gst_device_monitor_add_filter(m_deviceMonitor.get(), "Video/Source", NULL);
```
nullptr

jameshilliard replied:
Updated

jameshilliard committed:
[GStreamer][VideoCapture] Remove video/x-raw mime-type caps filters
https://bugs.webkit.org/show_bug.cgi?id=242256

Reviewed by Philippe Normand.

We support image/jpeg mime-type sources so having a video/x-raw
mime-type caps filter doesn't seem correct.

* Source/WebCore/platform/mediastream/gstreamer/GStreamerCaptureDeviceManager.cpp:
(WebCore::GStreamerCaptureDeviceManager::refreshCaptureDevices):

Canonical link: https://commits.webkit.org/252103@main
